src/source/core/Tasks.bs

```diff
@@ -364,6 +368,72 @@ namespace mc.tasks
     end if
   end function
 
+  function onTaskStateChange(state as string, task as mc.types.node)
+    isRunning = state = "run"
+    wasRunning = asBoolean(task._mIsRunning) = true
+    if not wasRunning and state = "run"
+      mc.setOrAddNodeField(task, "_mIsRunning", true)
+    else if wasRunning and state = "stop" or state = "done"
+      mc.setOrAddNodeField(task, "_mIsRunning", false)
+    end if
+
+    if isRunning
+      mc.tasks.trackTaskUsage(task)
+    else if wasRunning <> (task._mIsRunning = true)
+      mc.tasks.updateTaskUsage(task, wasRunning)
+    end if
+  end function
+
+  function trackTaskUsage(task as mc.types.node)
```

Comment:
I propose we simplify this a bit (the update and track task code seems a touch long and duplicated to me).
I also propose we create a new maestroConfig object, so that we don't fill up global with flags. let's move any global maestro flags to this.

something like this..

```
function onTaskStateChange(state as string, task as mc.types.node) as void
    isRunning = state = "run"
    wasRunning = asBoolean(task._mIsRunning) = true
    if not wasRunning and state = "run"
      mc.setOrAddNodeField(task, "_mIsRunning", true)
    else if wasRunning and state = "stop" or state = "done"
      mc.setOrAddNodeField(task, "_mIsRunning", false)
    end if

    if isRunning = wasRunning
      return
    end if

    globalObj = mc.getGlobal()
    taskType = task.subType()
    isLoggingTasks = asBoolean(globalObj.maestroConfig.isLoggingTaskTracking)

    maestroTasks = globalObj._maestroTasks = invalid

    if maestroTasks = invalid
      maestroTasks = mc.createSGNode("Node")
      mc.setOrAddNodeField(globalObj, "_maestroTasks", maestroTasks)
    end if
    numberRunning = asInteger(maestroTasks[taskType]) + isRunning ? 1 : 0

    if numberRunning > 0
      mc.setOrAddNodeField(maestroTasks, taskType, numberRunning)
    else
      maestroTasks.removeField(taskType)
    end if

    if isLoggingTasks
      fields = maestroTasks.getFields()
      for each field in fields.items()
        if not mc.isInteger(field.value)
          fields.delete(field.key)
        end if
      end for
    end if
  end function
  ```